modules/mongodb/mongodb.go

```diff
@@ -129,14 +131,77 @@ func (c *MongoDBContainer) ConnectionString(ctx context.Context) (string, error)
 	return c.Endpoint(ctx, "mongodb")
 }
 
-// eval builds an mongosh|mongo eval command.
-func eval(command string, args ...any) []string {
-	command = "\"" + fmt.Sprintf(command, args...) + "\""
+func setupEntrypointForAuth(req *testcontainers.GenericContainerRequest) {
+	req.Files = append(
+		req.Files, testcontainers.ContainerFile{
+			Reader:            bytes.NewReader(entrypointContent),
+			ContainerFilePath: entrypointPath,
+			FileMode:          0o755,
+		},
+	)
+	req.Entrypoint = []string{entrypointPath}
+	req.Env["MONGO_KEYFILE"] = keyFilePath
+}
+
+func configureRequestForReplicaset(
+	username string,
+	password string,
+	replicaSet string,
+	genericContainerReq *testcontainers.GenericContainerRequest,
+) error {
+	if !(username != "" && password != "") {
+		return noAuthReplicaSet(replicaSet)(genericContainerReq)
+	}
+	return withAuthReplicaset(replicaSet, username, password)(genericContainerReq)
+}
 
-	return []string{
-		"sh",
-		"-c",
-		// In previous versions, the binary "mongosh" was named "mongo".
-		"mongosh --quiet --eval " + command + " || mongo --quiet --eval " + command,
+func noAuthReplicaSet(replSetName string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) error {
+		cli := newMongoCli("", "")
+		req.Cmd = append(req.Cmd, "--replSet", replSetName)
+		initiateReplicaSet(req, cli, replSetName)
+
+		return nil
+	}
+}
+
+func initiateReplicaSet(req *testcontainers.GenericContainerRequest, cli mongoCli, replSetName string) {
+	req.WaitingFor = wait.ForAll(
+		req.WaitingFor,
+		wait.ForExec(cli.eval("rs.status().ok")),
+	).WithDeadline(60 * time.Second)
+	req.LifecycleHooks = append(
```

Comment:
Personal Nit: adding an empty line helps in readability in this block. Probably applicable to other places 🙏

Reply:
I will do the final touchup.